sfge/src/main/java/com/salesforce/cli/CacheCreator.java

```diff
@@ -0,0 +1,112 @@
+package com.salesforce.cli;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.salesforce.config.SfgeConfig;
+import com.salesforce.config.SfgeConfigProvider;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/** Creates cache data that can be used to execute Delta runs. */
+public final class CacheCreator {
+    private static final Logger LOGGER = LogManager.getLogger(CacheCreator.class);
+
+    private Dependencies dependencies;
+
+    public CacheCreator() {
+        this.dependencies = new Dependencies();
+    }
+
+    CacheCreator(Dependencies dependencies) {
+        this.dependencies = dependencies;
+    }
+
+    /**
+     * Creates cache data that can be used for future delta runs
+     *
+     * @param result from the current execution
+     */
+    public void create(Result result) {
+        createFilesToEntriesCache(result.getFilesToEntriesMap());
+    }
+
+    private void createFilesToEntriesCache(FilesToEntriesMap filesToEntriesMap) {
+        final String jsonData = filesToEntriesMap.toJsonString();
+        final String filesToEntriesDataFilename = getFilesToEntriesDataFilename();
+
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info(
+                    "About to create cache for filesToEntriesMap at " + filesToEntriesDataFilename);
+        }
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(
+                    "Content to be written in " + filesToEntriesDataFilename + ": " + jsonData);
+        }
+
+        try {
+            dependencies.writeFile(filesToEntriesDataFilename, jsonData);
+        } catch (IOException e) {
+            // Putting a warning instead of stopping the process since this is not in the critical
+            // path
+            if (LOGGER.isWarnEnabled()) {
+                LOGGER.warn("Exception occurred while writing filesToEntriesMap cache data.", e);
+            }
+        }
+    }
+
+    private String getFilesToEntriesDataFilename() {
+        final SfgeConfig sfgeConfig = SfgeConfigProvider.get();
+        final String cacheDir = sfgeConfig.getCacheDir();
+        final String filesToEntriesCacheData = sfgeConfig.getFilesToEntriesCacheData();
+        return cacheDir + File.separator + filesToEntriesCacheData;
+    }
+
+    static class Dependencies {
+
+        /**
+         * Write to file. The current assumption is to overwrite the file. TODO: This could be
```

Comment:
Sounds good. I can add the new flag to skip caching.

Reply:
@jag-j If the run is always based on the full-run, isn't there a probability of a previously fixed violation to show up to Developer simply because the entry method was not invoked? We may need to think through this some more.

Reply:
@rmohan20 and I discussed offline about this and not updating the cache for delta runs is the right thing to do right now.